feat(HACBS-1308) : add a deny all network policy

naftalysh opened the pull request:
Applied PR as per https://issues.redhat.com/browse/HACBS-1308 - Put a network policy around release-service controller, prevent egress to everywhere

naftalysh commented:
/retest

naftalysh commented:
/retest

naftalysh commented:
/retest

naftalysh commented:
/retest-required

naftalysh commented:
/retest

naftalysh commented on config/manager/network_policy.yaml:
```diff
@@ -0,0 +1,12 @@
+    matchLabels:
+      control-plane: controller-manager
+  policyTypes:
+    - Egress
```
Adding

naftalysh replied:
Added

naftalysh commented:
> Looks good to me apart from the comments. One thing you should do though is to change the commit title to:
> 
> ```
> feat(HACBS-1308): add a deny all network policy
> ```
> 
> The commit needs as well to be GPG signed and have a Sign-off line.

Done

naftalysh commented:
> Looks good to me apart from the comments. One thing you should do though is to change the commit title to:
> 
> ```
> feat(HACBS-1308): add a deny all network policy
> ```
> 
> The commit needs as well to be GPG signed and have a Sign-off line.

Done

davidmogar commented:
Given that we don't have a network policy yet, should both of your PR be merged together? @johnbieren comment?

johnbieren commented:
What is the binary file that is now in the diff @naftalysh ?

johnbieren commented:
> Given that we don't have a network policy yet, should both of your PR be merged together? @johnbieren comment?

I am fine with either way. The other one would have to be rebased if it is one then the other obviously, but whatever is easiest for @naftalysh. I think each PR adds some useful functionality so it is okay to merge one then the other if that makes things easier

naftalysh commented:
> What is the binary file that is now in the diff @naftalysh ?

It shouldn't be there - its a product of gitlint -d command by which I checked the commit validity
I'll take it out of there

naftalysh commented:
> > Given that we don't have a network policy yet, should both of your PR be merged together? @johnbieren comment?
> 
> I am fine with either way. The other one would have to be rebased if it is one then the other obviously, but whatever is easiest for @naftalysh. I think each PR adds some useful functionality so it is okay to merge one then the other if that makes things easier

1306 overrides 1308 because its too blocks all egress, but in any case I shall merge both

naftalysh commented:
/cc @davidmogar

naftalysh commented:
/CC @davidmogar

davidmogar commented:
Could you squash both commits before merging, @naftalysh?

davidmogar commented:
After talking with @naftalysh we have agreed on this change needing to be tested. The happy test passing doesn't guarantee the change works as we don't have any egress so far. We need to attach to the running container and try to access anything outside (avoiding DNS as probably this is not properly configured in the container itself).

naftalysh commented:
Suggested edit:
```diff
diff --git a/.github/gitlint/__pycache__/contrib_format_conventional_commits.cpython-311.pyc b/.github/gitlint/__pycache__/contrib_format_conventional_commits.cpython-311.pyc
new file mode 100644
index 0000000..eaa4cb0
Binary files /dev/null and b/.github/gitlint/__pycache__/contrib_format_conventional_commits.cpython-311.pyc differ

```

naftalysh commented:
> After talking with @naftalysh we have agreed on this change needing to be tested. The happy test passing doesn't guarantee the change works as we don't have any egress so far. We need to attach to the running container and try to access anything outside (avoiding DNS as probably this is not properly configured in the container itself).

[nshprai]
Just did the following:
1. According to the https://github.com/naftalysh/release-service/tree/main#build-and-push-a-new-image procedure 
    1a. Updated the Dockerfile second FROM image to take my own prepared image that has client tools and oc
           FROM quay.io/nshprai/fedora-35:latest
    1b. Followed the procedure to build the release-service image - 
           quay.io/nshprai/release-service:294d0aac6cd2097f454e38b34222fbcda7af5677
           
2. Executed infra-deployments into a new provisioned Hive cluster (version 4.11.20)
    2a. Updated the infra-deployments preview.env configuration file to have the following configuration:
           export RELEASE_IMAGE_REPO=quay.io/nshprai/release-service
           export RELEASE_IMAGE_TAG=294d0aac6cd2097f454e38b34222fbcda7af5677
           export RELEASE_RESOURCES=https://github.com/naftalysh/release-service/config/default?ref=294d0aac6cd2097f454e38b34222fbcda7af5677
           
    2b. Executed infra-deployments procedure into a new Hive provisioned cluster which succeeded.
    
3. tested the happy-path e2e-tests which succeeded again
4. Check the http://www.microsoft.com:80 site info
     4a. Tried to check from outside the cluster with curl command - the http://www.microsoft.com:80 status
               curl -m 15 -L -I http://www.microsoft.com:80                                                                                                                                                                                             
               ─╯
               HTTP/1.1 200 OK
               
     4b. Found the site's IP address with the following command - ping  www.microsoft.com
           ping  www.microsoft.com                                                                                                                                                                                                                  
           ─╯
           PING e13678.dscb.akamaiedge.net (95.100.206.108) 56(84) bytes of data.
           
           So we know the sites address (95.100.206.108)
           
5. Performed the following commands within the release-service pod manager container
    5a. # Get release-service pod
          POD_NAME=$(oc get pods -n release-service | awk '{print $1}' | tail -n 1)
          
    5b. Tried to check from outside the cluster with curl command - the http://www.microsoft.com:80 status 
           (15 sec timeout)
               oc exec $POD_NAME -n release-service -c manager  -- curl -m 15 -L -I http://www.microsoft.com:80
               ─╯
               % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
                0     0    0     0    0     0      0      0 --:--:--  0:00:20 --:--:--     0
          curl: (28) Resolving timed out after 15000 milliseconds
          command terminated with exit code 28
          
          ---Failure---
               
     5c. Tried the same with the site's IP address (95.100.206.108)
           oc exec $POD_NAME -n release-service -c manager  -- curl -m 15 -L -I 95.100.206.108:80                                                                                                                                                                                                                 
           ─╯
           curl: (28) Connection timed out after 15001 milliseconds
           command terminated with exit code 28
           
            ---Failure---
            
      5d. I don't have a ping command in the pod so I cannot ping the site's address but that's not needed.
      
      
      Conclusion - The pod is blocked for Egress communications

naftalysh commented:
/CC @davidmogar 
**We can't merge this PR before I undo the Dockerfile update which uses my image!**

davidmogar commented on config/manager/network_policy.yaml:
```diff
@@ -0,0 +1,13 @@
+      control-plane: controller-manager
+  policyTypes:
+    - Egress
+---
```
Remove this line please.

naftalysh replied:
sure

naftalysh replied:
Done

davidmogar reviewed (commented):
Could you please squash the commits?

naftalysh commented:
/retest

naftalysh commented:
/restest

naftalysh commented:
/retest

naftalysh commented:
> Could you please squash the commits?

Done

kasemAlem commented:
@davidmogar , how this can be merged ?

naftalysh commented:
/retest-needed
